src/task.rs

```diff
@@ -40,8 +40,15 @@ pub struct TaskResult {
 fn read_depfile(path: &Path) -> anyhow::Result<Vec<String>> {
     let mut bytes = match std::fs::read(path) {
         Ok(b) => b,
+        Err(e) if e.kind() == std::io::ErrorKind::NotFound => Vec::new(),
```

Comment:
I think this would be better if this returned here, so we don't need to use bytes.len() as a marker for missing file.